src/Manager/ShapeManager.ts

```diff
@@ -13,15 +13,22 @@ import AbstractObservable from '../Observer/AbstractObservable';
 import IAction from '../Action/IAction';
 import ActionType from '../Action/ActionType';
 import IDocumentEventHandler from './IDocumentEventHandler';
+import Canvas from '../Canvas';
 
 // eslint-disable-next-line max-len
-class ShapeManager extends AbstractObservable<IAction> implements IObserver<IAction>, ICanvasEventHandlder, IDocumentEventHandler {
-  factory: ShapeFactory = new ShapeFactory();
+export default class ShapeManager extends AbstractObservable<IAction> implements IObserver<IAction>, ICanvasEventHandlder, IDocumentEventHandler {
+  factory: IFactory<IShapeInfo, Shape> = new ShapeFactory();
   shapes: Array<Shape> = [];
   undoShapes: Array<Shape> = [];
   currentShape: Shape | null = null;
   basePoint: Point | null = null;
   isDrawing = false;
+  canvas: Canvas;
+
+  constructor(cnvs: Canvas) {
```

Comment:
cnvs ? Carpe Normande Vivant Saâcy-sur-marne (77730) ?
Ca coute rien d'écrire canvas et c'est tellement plus clair.
Je comprend que c'est pour pas avoir `this.canvas = canvas;` mais je trouve pas ca gênant

Reply:
Nope, I completely agree with you, but there was a linter rule acting weird on my part, so I had to work around. The rule does not apply anymore now, so I can name the variable canvas.